sdk/couchbase/tests/user_management.rs

```diff
@@ -0,0 +1,297 @@
+        ];
+
+        let group = Group::new(&group_name, desc, roles);
+        create_group(&cluster.users(), group).await;
+
+        delete_group(&cluster.users(), &group_name).await;
+
+        let opts = GetGroupOptions::new();
+        try_until(
+            Instant::now().add(Duration::from_secs(30)),
+            Duration::from_millis(500),
+            "get group after delete did not fail in time",
+            async || {
+                let err = match cluster.users().get_group(&group_name, opts.clone()).await {
+                    Ok(_) => return Ok(None),
+                    Err(e) => e,
+                };
+
+                if err.msg.to_ascii_lowercase().contains("not found") {
+                    return Ok(Some(()));
+                }
+
+                Ok(None)
+            },
+        )
+        .await;
+    });
+}
+
+#[test]
+fn test_get_group() {
+    run_test(async |cluster| {
+        let group_name = new_key();
+        let desc = new_key();
+        let roles = vec![
+            Role::new("replication_target").bucket(&cluster.default_bucket),
+            Role::new("replication_admin"),
+        ];
+
+        let group = Group::new(&group_name, desc, roles);
+        create_group(&cluster.users(), group.clone()).await;
+
+        let opts = GetGroupOptions::new();
+
+        let actual = try_until(
+            Instant::now().add(Duration::from_secs(30)),
+            Duration::from_millis(500),
+            "failed to get group in time",
+            async || match cluster.users().get_group(&group_name, opts.clone()).await {
+                Ok(g) => Ok(Some(g)),
+                Err(e) => Err(e),
+            },
+        )
+        .await;
+
+        assert_eq!(actual.name, group.name);
+        assert_eq!(actual.roles, group.roles);
+        assert_eq!(actual.description, group.description);
+        assert_eq!(actual.ldap_group_reference, group.ldap_group_reference);
+    });
+}
+
+#[test]
+fn test_get_all_groups() {
+    run_test(async |cluster| {
+        let group_name = new_key();
+        let desc = new_key();
+        let roles = vec![
+            Role::new("replication_target").bucket(&cluster.default_bucket),
+            Role::new("replication_admin"),
+        ];
+
+        let group = Group::new(&group_name, desc, roles);
+        create_group(&cluster.users(), group.clone()).await;
+
+        let opts = GetAllGroupsOptions::new();
+
+        let actual: Group = try_until(
+            Instant::now().add(Duration::from_secs(30)),
+            Duration::from_millis(500),
+            "failed to get group in time",
+            async || {
+                let groups = match cluster.users().get_all_groups(opts.clone()).await {
+                    Ok(g) => g,
+                    Err(e) => return Err(e),
+                };
+
+                for actual_group in groups {
+                    if actual_group.name == group_name {
+                        return Ok(Some(actual_group));
```

Comment:
[`Ok(groups.find(...))`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find)?

Reply:
I like it.